app/models/akka/Messages.scala

```diff
@@ -128,4 +128,46 @@ object Messages {
    * Sent to Downstream actor to remove disconnected DSLinks.
    */
   case object RemoveDisconnectedDSLinks
+
+  /**
+    * Sent to StateKeeper to get subscription stream ref
+    */
+  case class GetSubscriptionSource()
+
+
+  /**
+    * Sent from StateKeeper with subscription stream ref
+    */
+  case class SubscriptionSourceMessage(sourceRef: SourceRef[DSAMessage])
+
+  /**
+    * Sent to StateKeeper to add subscription message
+    * @param message
+    */
+  case class PutNotification(message:SubscriptionNotificationMessage)
```

Comment:
It makes things longer, but I would rename it to include "Subscription", otherwise it just sounds too generic. Alternatively, if this and other messages below only pertain to StateKeeper, perhaps move them into `object StateKeeper`?

Reply:
moved to QoSState